### ISSUE TITLE
Why do you save and restore old model's weights?

### ISSUE THREAD

**Issue by envytails** [S-Choudhuri/Adversarial-Domain-Adaptation-with-Keras#3]
Hi! thank you for your source code.
There is something that is not clear to me. For example, why during the training do you save weights and restore them after the training?

**Comment by S-Choudhuri** [comment]
Hi! This is because the feature extractor parameters (in build_embedding) is trained through two channels, (1) one from the classifier and the (2) other from the discriminator (discriminator loss is reversed into the feature extractor during back propagation by flipping domain labels). The weights are saved in the midway to ensure that (a) the discriminator part is not trained twice in one epoch and (2) the feature extractor layers learn exclusively from the loss during : (i) training conducted by label flipping (gradient ascent) in the discriminator-end and not by the regular gradient descent (ii) back-propagation of the classifier loss.
